### PR TITLE
ManuallyDrop != MaybeUninit

### PR DESCRIPTION
**Pull request by RalfJung**
Cc @Centril

### CODE REVIEW

File: src/libcore/mem.rs [hunk 1]
```diff
@@ -900,10 +900,16 @@ pub fn discriminant<T>(v: &T) -> Discriminant<T> {
...
 
+// FIXME: Reference `MaybeUninit` from these docs, once that is stable.
 /// A wrapper to inhibit compiler from automatically calling `T`’s destructor.
 ///
 /// This wrapper is 0-cost.
 ///
+/// `ManuallyDrop<T>` is subject to the same layout optimizations as `T`.
+/// As a consequence, it has *no effect* on the assumptions that the compiler makes
+/// about all values being initialized at their type.  In particular, initializing
+/// a `ManuallyDrop<&T>` with [`mem::zeroed`] is undefined behavior.
```
Comment (nagisa) [inline src/libcore/mem.rs]: It is very easy to miss `&` near that `T`. I spent a few minutes being confused. Can we give a better, more-in-your-face, example here? Like `ManuallyDrop<NonNull<T>>`?

Comment (Centril) [reply src/libcore/mem.rs]: Maybe `ManuallyDrop<&mut T>`?

Comment (RalfJung) [reply src/libcore/mem.rs]: I like `&mut T`.

File: src/libcore/mem.rs [hunk 2]
```diff
@@ -935,6 +941,8 @@ pub fn discriminant<T>(v: &T) -> Discriminant<T> {
 ///     }
 /// }
 /// ```
+///
+/// [`mem::zeroed']: fn.zeroed.html
```
Comment (Centril) [inline src/libcore/mem.rs]:
```suggestion
/// [`mem::zeroed`]: fn.zeroed.html
```

**Comment by Centril** [comment]
Failed in https://github.com/rust-lang/rust/pull/58798#issuecomment-468130877, @bors r-

**Comment by RalfJung** [comment]
Why did the PR CI not catch that...?

**Pull request merged**